<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><head>



  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <title>snp.plotter</title>


  <link href="snp.css" rel="stylesheet" type="text/css">

</head><body>

<div id="overallbox">
<a name="global_nav"></a>
<div id="logo"><a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter.html">snp.plotter</a></div>

<div id="nav"> 
<li><a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter.html">Introduction</a></li>
<li><a href="#usage">Usage</a></li>
<li><a href="#downloads">Download</a></li>
<li><a href="#rpad">Web Interface</a></li>
<li><a href="#file_formats">File Formats</a></li>
<li><a href="#author">Author</a></li>
</div>

<div class="box">
<h1><a name="introduction"></a>What is snp.plotter?</h1>

<p><a href="#downloads"><img class="floatRight" alt="sp.download.png" src="sp.png"></a>
snp.plotter is an R package that creates publishable-quality plots of
p-values using single SNP and/or haplotype data. Main features of the
package include options to display a linkage disequilibrium (LD) plot
and the ability to plot multiple sets of results simultaneously. Plots
can be created using global and/or individual haplotype p-values along
with single SNP p-values. Images are created
as either <a href="http://en.wikipedia.org/wiki/Pdf">Portable Document Format (PDF)</a> or <a href="http://en.wikipedia.org/wiki/Encapsulated_PostScript">Encapsulated (EPS)</a> files.</p>

<h1>What do snp.plotter images look like?</h1>
<p>Here is a sample image as both a <a href="http://cbdb.nimh.nih.gov/%7Ekristin/null_rsq_phys_color.pdf">PDF</a> and a <a href="http://cbdb.nimh.nih.gov/%7Ekristin/null_rsq_phys_color.eps">EPS</a> image. PDF images can be viewed using the free <a href="http://www.adobe.com/products/acrobat/readstep2.html">Adobe Reader</a> software. EPS (or PDF) images can be viewed using front-end software for <a href="http://www.cs.wisc.edu/%7Eghost/">Ghostscript</a>, a Postscript and PDF interpreter: <a href="http://www.cs.wisc.edu/%7Eghost/">Ghostview and GV</a> for Unix/X11 systems and <a href="http://www.cs.wisc.edu/%7Eghost/">GSView</a> for Windows. 

<img alt="null_rsq_phys_color.png" src="null_rsq_phys_color.png">

</p>
<h1>Who are the intended users of snp.plotter?</h1>

<p>Researchers conducting gene association studies who need to create
images showing results in the form of p-values of their work along with
a <a href="http://en.wikipedia.org/wiki/Linkage_disequilibrium">linkage disequilibrium</a>
plot to show the underlying structure of the study population. Because
snp.plotter is a visualization tool, the type of study design used
either family-based, case-control, or a combination is irrelevant.
</p>

<h1>What are the main features of snp.plotter?</h1>

<p>
	</p><ul>
		<li>Publishable-quality graphics in resolution-independent formats</li>
<li>Free and open source</li>
<li>Visualization of global or individual haplotype p-values</li>
<li>Visualization of results either physically spaced or evenly spaced</li>
<li>Ability to plot multiple sets of results</li>
<li>Optional web interface</li>
<li>Customizable parameters</li>
	<ul>
	<li>Labels</li>
	<li>Symbols</li>
	<li>Colors/color schemes</li>
	<li>LD metric</li>
	<li>P-value threshold</li>
	</ul>

	</ul>
</p>

<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="up_arrow.png"></a>
</div>
<div class="box">

<h1><a name="usage"></a>How do I install snp.plotter?</h1>

<p>snp.plotter requires the installation of R, the statistical computing software, freely available <a href="http://www.r-project.org/">here</a>
for Linux, Windows, or MacOS. snp.plotter uses three contributed
packages: grid, lattice, and genetics available from the R Project
website. snp.plotter can be downloaded <a href="#downloads">here.</a> Packages in the current working directory of R can be installed using the <strong>install.packages</strong> command. 

</p><p>In <strong>Windows,</strong> make sure that you have downloaded the .zip version of a package, which is a pre-compiled binary. Change the word <strong>PACKAGE_FILENAME</strong> to the name of the package being installed.
</p><pre>install.packages("PACKAGE_FILENAME.zip", repos=NULL)</pre>

In <strong>Unix/Linux,</strong> make sure that you have downloaded a source package, the .tar.gz version, of a package. Change the word <strong>PACKAGE_FILENAME</strong> to the name of the package being installed.
<pre>install.packages("PACKAGE_FILENAME.tar.gz", repos=NULL)</pre>

The following error indicates that additional packages need to be installed: <strong>PACKAGE_FILENAME</strong> can be grid or genetics. These packages can be downloaded from <a href="http://cran.r-project.org/src/contrib/PACKAGES.html">CRAN.</a> and installed using the same process described above.
<pre>Error: package 'PACKAGE_FILENAME' could not be loaded</pre>

<h1><a name="usage"></a>How is snp.plotter used?</h1>

Once snp.plotter and its dependencies are installed, snp.plotter can be loaded into R using this command:

<pre>library(snp.plotter)
</pre>

<p>snp.plotter is then run using the following command. Explanation of
snp.plotter configuration files and input files is provided in the <a href="#file_formats">file formats</a>
section. The configuration file specifies all the characteristics for
the plot to be produced and the names of the data files to be used as
input. Example <a href="#downloads">configuration and input files</a> are also provided. 

</p><pre>snp.plotter(config.file="config.txt")
</pre>

Information about the configuration options is provided in the <a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter.pdf">documentation,</a> which can viewed from within R using this command. 

<pre>?snp.plotter
</pre>


<h1>How do I incorporate the snp.plotter figures into my publication?</h1>

<p>The listing below shows the most common scenarios for inclusion of a
figure into a publication; methods will vary based on the intended
publication type:
</p>

<ol>
<li><strong>Inclusion of PDF into Word document:</strong> PDF images can be selected and copied using the Adobe Reader snapshot tool <img alt="snapshot_tool.png" src="snapshot_tool.png">. These images can then be pasted into a Word document. 

</li><li><strong>Inclusion of PDF/EPS into LaTeX document:</strong> Images included into DVI file using <strong>latex</strong> must be in the EPS format. PDF documents created by <strong>pdflatex</strong> cannot use files in the EPS format and requires the PDF format. The following is basic LaTeX code for including an image. <strong>FILENAME</strong> must be changed to the appropriate filename. 
<pre>\documentclass{article}

\usepackage{graphicx}

\begin{document}
\begin{figure}
\includegraphics{FILENAME}
\end{figure}
\end{document}
</pre> 
</li>

<li><strong>Conversion of PDF/EPS file into other formats using Ghostscript</strong>:
Ghostscript can be used to convert PDF and EPS files into a variety of
raster formats such as JPG, BMP, PNG, etc. The following commandline
converts an EPS image into a JPG image; the resulting image has a
resolution of 300 DPI and pixel dimension of 1050x1050 pixels. Further
information on the usage of Ghostscript can be found <a href="http://www.cs.wisc.edu/%7Eghost/doc/AFPL/8.00/Use.htm">here.</a> <strong>FILENAME</strong> must be changed to the appropriate filename. 
<pre>gs -q -dNOPAUSE -dSAFER -dBATCH -r300 -g1050x1050 -sOutputFile=FILENAME.jpg -sDEVICE=jpeg FILENAME.eps
</pre> 
</li>


</ol>
<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="up_arrow.png"></a>
</div>

<div class="box">
<h1><a name="rpad"></a>Are there other ways to interact with snp.plotter?</h1>

<p>There is an optional web interface for snp.plotter using <a href="http://www.rpad.org/Rpad/">Rpad</a> available for <a href="#downloads">download;</a> (<a href="http://cbdb.nimh.nih.gov/%7Ekristin/rpad_interface.png">screenshot</a>).
Rpad allows the creation of web interfaces for R so that end users do
not need to install anything on their computers. The web interface is
best suited to intranet environments since users have complete access
to any command in R and any system command. The instructions presented
here are for local installation and usage; snp.plotter must be
installed on the machine running Rpad. Instructions for server
deployment are presented on the <a href="http://www.rpad.org/Rpad/BasicDocumentation.html">Rpad website.</a></p> <p>Once installed Rpad is run using the following two commands:

</p><pre>library(Rpad)
Rpad()
</pre>

In Windows, the web interface (<strong>snp_plotter.Rpad</strong>) is placed in <strong>C:\Program Files\R\R-2.3.1\library\Rpad\basehtml</strong>
(location may vary depending on R version and where R was originally
installed). If Rpad is running, snp.plotter can then be accessed at <a href="http://127.0.0.1:8079/snp_plotter.Rpad">http://127.0.0.1:8079/snp_plotter.Rpad</a>
After setting the various options and clicking "Create Plot", users
will see various output messages; the last message provides the
location of where the created image can be found.
<p>The interface includes the majority of features, but is limited to
one dataset. The snp.plotter interface can be extended with basic
knowledge of HTML and R to manipulate options presented or to perform
additional analysis the researcher may require.</p>


<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="up_arrow.png"></a>
</div>

<div class="box">
<h1><a name="file_formats"></a>What file types are used by snp.plotter?</h1>

<p>snp.plotter uses tab-delimited text files as input files. These
files can be created in a basic text editor such as Notepad in Windows
or Excel saved as tab-delimited files using the "Save As" option. A
dataset is composed a configuration file, a SNP and haplotype file for
each result set, one genotype file, and an optional palette file. More
information about these files can be found in the <a href="#downloads">documentation.</a>
</p><ul>

  <li><strong>Configuration file:</strong> The configuration
file is the preferred method of running snp.plotter; it allows users to
save preferred settings and avoids the inconvenience of writing
extended R commands, repeatedly. There cannot be spaces after commas. A
complete description of all possible plot characteristics can be found
in the <a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter.pdf">documentation.</a>
Result sets without haplotype results should be placed after those with
results; no HAP.FILE entry should be specified in this case. <pre> 
SNP.FILE=snp8_ss.txt,snp8_ss2.txt
HAP.FILE=snp8_haplo.txt,snp8_haplo2.txt
PALETTE.FILE=palette.txt
EVEN.SPACED=TRUE
USE.GBL.PVAL=TRUE
DISP.HAP=TRUE
DISP.LDMAP=FALSE
LD.COLOR.SCHEME=heat
COLOR.LIST=blue,red
SYMBOLS=circle,NA
PVAL.THRESHOLD=1
LAB.Y=ln
GENOTYPE.FILE=snp8_geno.txt
LD.TYPE=rsquare
DISP.COLOR.BAR=TRUE
DISP.TYPE=symbol
DISP.LEGEND=TRUE
SAMPLE.LABELS=Sample 1,Sample 2
IMAGE.TYPE=eps
DISP.SNP=TRUE
IMAGE.TITLE="8 SNP Example"
IMAGE.NAME=snp8_example
IMAGE.SIZE=3.5
	</pre>
 </li>

  <li><strong>SNP.FILE:</strong> SNP.FILE includes four necessary columns <strong>ASSOC, SNP.NAME, LOC,</strong> and <strong>SS.PVAL</strong>
corresponding to positive or negative association (indicating
protective or susceptibility alleles, a SNP label, the location, and a
p-value for each SNP. SNP labels may not start with numbers. In the
figure, SNPs are indicated by the symbols chosen for the data, if
symbol type 'NA' is specified, the SNP.FILE <strong>ASSOC</strong>
column is read and an up-triangle and down-triangle are used to
indicate positive and negative association (indicating susceptibility
or protective alleles), respectively. <pre> 
ASSOC	SNP.NAME	LOC	SS.PVAL
+	rs10_8	126272509	0.065
-	rs11_8	126274467	0.029
+	rs12_8	126275017	0.046
-	rs13_8	126275750	0.005
+	rs14_8	126283659	0.258
+	rs15_8	126284467	0.036
-	rs16_8	126298007	0.05
-	rs17_8	126299520	0.202
	</pre>
 </li>

  <li><strong>HAP.FILE:</strong> HAP.FILE includes three necessary columns <strong>ASSOC, G.PVAL,</strong> and <strong>I.PVAL</strong>
corresponding to positive or negative association (indicating
protective or susceptibility alleles, a global p-value and an
individual p-value for each haplotype followed by a set of columnns of
SNPs with corresponding haplotypes. Haplotypes are presented in a
step-wise fashion with the major allele given as 1 and the minor allele
as 2; haplotype variants for a set of SNPs should be grouped. SNP
labels in <strong>HAP.FILE</strong> must be the same as in <strong>SNP.FILE,</strong>
and only SNPs with corresponding haplotypes need to be included. In the
figure, unfilled symbols connected by solid lines are used to indicate
global haplotype p-values, (a circle is used if no symbol is specified
for the dataset). Unfilled and filled symbols are used to indicate
alleles 1 and 2, respectively connected by solid lines and dashed lines
for positive and negative association (indicating susceptibility or
protective haplotypes) when using indivudal haplotype p-values. <pre> 
ASSOC	G.PVAL	I.PVAL	rs10_8	rs11_8	rs12_8	rs13_8	rs14_8	rs15_8	rs16_8	rs17_8
-	0.015	0.004	1	1	1					
+	0.015	0.062	1	2	2					
+	0.075	0.079		1	1	1				
+	0.075	0.039		2	2	2				
-	0.032	0.121			1	1	1			
+	0.032	0.153			1	2	2			
+	0.425	0.474				1	1	1		
+	0.425	0.003				2	2	2		
+	0.1	0.077					1	1	1	
+	0.1	0.1					1	2	2	
-	0.003	0.341						1	1	1
+	0.003	0.001						2	2	2
	</pre></li>

  <li><strong>GENOTYPE.FILE:</strong>
GENOTYPE.FILE is a modified Linkage PED file. Each row should have the
following information: family ID, individual ID, father ID, mother ID,
sex, and affection status followed by marker loci coded as binary
factors, as shown in the example below. This file should not have
column headers. <pre> 
1	1	0	0	1	1	1	1	1	1
2	1	0	0	2	1	1	1	1	1
3	1	0	0	1	1	1	1	1	2
4	1	0	0	2	1	1	1	2	2
5	1	0	0	1	1	0	0	1	1
	</pre>
 </li>

  <li><strong>PALETTE.FILE (OPTIONAL):</strong> PALETTE.FILE can be used to specify a color palette for the LD heatmap. To use the the PALETTE.FILE option, <strong>LD.COLOR.SCHEME</strong>
must be set to "custom" and a file name must be provided to the
PALETTE.FILE option. Custom color schemes cannot be used in conjunction
with predefined color schemes such as "heat" or "gray". PALETTE.FILE
colors are <a href="http://en.wikipedia.org/wiki/X11_color_names">hexidecimal HTML color codes;</a> one color per line. The first and last colors correspond to the lowest and highest value of the chosen LD metric, respectively.
	<pre> 
FFFFFF
CCCCFF
9999FF
6666FF
3333FF
0000FF
0000CC
000099
000066
	</pre>
 </li>
</ul>


<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="up_arrow.png"></a>
</div>

<div class="box">

<h1><a name="downloads"></a>Where can I get the source code of snp.plotter for my own project?</h1>

<p>
</p><ul>

  <li><a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter_0.2.tar.gz">Download snp.plotter as an R source package for Unix/Linux</a></li>
  <li><a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter_0.2.zip">Download snp.plotter for Windows</a></li>
  <li><a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp_plotter.Rpad">Download snp.plotter Rpad Interface</a></li>
  <li>Download example configuration and input files: <a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter.20SNP.dataset.zip">20 SNP</a> dataset</li>
  <li><a href="mailto:?Subject=Interesting%20R%20package:%20snp.plotter&amp;Body=Hi%2C%0D%0A%0D%0AI%20think%20you%27ll%20find%20the%20snp.plotter%20R%20package%20interesting,%0D%0Ahttp://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter.html%20It%20creates%20publishable-quality%20images%20of%20gene%20association%20results.%20Check%20it%20out%21%0D%0A%0D%0ARegards%2C%0D%0A">Tell a friend</a></li>
  <li><a href="http://www.r-project.org/">Learn about the R project for statistical computing</a></li>
  <li><a href="http://cbdb.nimh.nih.gov/%7Ekristin/snp.plotter.pdf">Read the documentation</a></li>
  <li><a href="http://cbdb.nimh.nih.gov/%7Ekristin/changelog.txt">View release notes</a></li>
  <li><a href="mailto:augustin@mail.nih.gov?subject=snp.plotter%20Bug%20Report">Report a bug</a></li>

</ul>

<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="up_arrow.png"></a>
</div>

<div class="box">
<h1><a name="author"></a>Authors</h1>
<p>
Augustin Luna, BS<br>
2005-2007: Postbaccalaureate Intramural Research Training Award Fellow<br>
Clinical Brain Disorders Branch (CBDB)/Genes, Cognition, and Psychosis
Program (GCAP), National Institute of Mental Health (NIMH), National
Institutes of Health (NIH) <br><br>
2007-Current: Pre-Doctoral Intramural Research Training Award Fellow<br>
Center for Cancer Research (CCR), Laboratory of Molecular Pharmacology (LMP), National Cancer Institute (NCI), National
Institutes of Health (NIH) <br><br>
BS, Biomedical Engineering, Georgia Institute of Technology <br>
PhD candidate, Bioinformatics, Boston University<br>
</p>
and
<p>
<a href="http://cbdb.nimh.nih.gov/%7Ekristin">Kristin Nicodemus, Ph.D., M.P.H.</a>
<br>
Statistical Geneticist/Genetic Epidemiologist<br>
Genes, Cognition and Psychosis Program, IRP, NIMH, NIH</p>
Acknowledgements
<p>
Anushka Aqil - For her feedback on the software<br>
CBDB/GCAP, NIMH, NIH <br><br>
Brandeis University</p>
<p>
<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="up_arrow.png"></a>
</p></div>
</div>

<!-- Site Meter -->
<a id="idSiteMeterHREF" href="http://www.sitemeter.com/stats.asp?site=s29snpplotter" target="_top"><img src="meter.gif" alt="Site Meter" border="0"></a><iframe style="display: none;" src="a.html" frameborder="0" height="0" width="0"></iframe><script type="text/javascript" src="counter.js">
</script>
<noscript>
<a href="http://s29.sitemeter.com/stats.asp?site=s29snpplotter" target="_top">
<img src="http://s29.sitemeter.com/meter.asp?site=s29snpplotter" alt="Site Meter" border="0"/></a>
</noscript>
<!-- Copyright (c)2006 Site Meter -->

</body></html>